crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -535,6 +552,29 @@ impl AuthorityStorePruner {
         Ok(Some(sst_file))
     }
 
+    fn pruning_tick_duration_ms(epoch_duration_ms: u64) -> u64 {
+        min(epoch_duration_ms / 2, 60 * 1000)
+    }
+
+    fn smoothed_max_eligible_checkpoint_number(
+        pruned_checkpoint: CheckpointSequenceNumber,
+        max_eligible_checkpoint: CheckpointSequenceNumber,
+        epoch_duration_ms: u64,
+    ) -> CheckpointSequenceNumber {
+        if max_eligible_checkpoint == 0 {
+            return max_eligible_checkpoint;
+        }
+        let num_intervals = epoch_duration_ms
+            .checked_div(Self::pruning_tick_duration_ms(epoch_duration_ms))
+            .unwrap_or(1);
+        let delta = max_eligible_checkpoint
+            .checked_sub(pruned_checkpoint)
+            .unwrap_or_default()
+            .checked_div(num_intervals)
+            .unwrap_or(1);
```

Comment:
this doesn't seem to account for progress. won't it always schedule a fixed fraction of the remaining checkpoints, rather than (as I would have expected) a fixed fraction of the total number of checkpoints in the epoch?

Reply:
it's not guaranteed to be fully uniformed, but on every interval a new bunch of checkpoints will be produced by the consensus, so right edge of the interval for calculation is also moving.
`delta = (last_executed_checkpoint - last_pruned_checkpoint) / number_of_tick_intervals_per_epoch`.
(i.e. `right edge != last_executed_checkpoint_in_previous_epoch`)